### PR TITLE
Integrate InAppPurchaseStore with networking

### LINKED ISSUE #7836: [IAP] Integrate StoreKit 2 implementation with networking
Once the backend is implemented and set up, stop using mock products, and integrate our StoreKit code from #7834 with the networking code from #7835 to get the available product identifiers from the API and record any purchases in the server.

### PR DESCRIPTION
**Pull request by koke**
Closes: #7836

### Description

This PR makes the `InAppPurchaseStore` actually talk to our servers to get the list of products and report any purchases. I left a `useBackend` property in the store to quickly switch the behavior back to local-only, since the endpoints aren't working yet.

### Testing instructions

This PR still uses a StoreKit configuration file by default. This makes it possible to test on simulator, but the receipts won't validate with Apple's servers. On the other hand, using the IAP sandbox servers will work for validation, but testing has to be done on a real device and requires [some extra setup](https://developer.apple.com/documentation/storekit/in-app_purchase/testing_in-app_purchases_with_sandbox). To change whether a StoreKit configuration file is used, in Xcode go to Product > Scheme > Edit Scheme, and change the setting in the Options tab.
Screen Shot 2022-10-20 at 07 34 34

More documentation on [testing IAP with sandbox here](https://developer.apple.com/documentation/storekit/in-app_purchase/testing_in-app_purchases_with_sandbox).

1. Ensure In-app purchases are enabled in Settings > Experimental features
2. Go to the Hub menu and tap "[Debug] IAP"
3. The debug screen should show a list of products with one product (currently the request will fail to get products)
4. Tapping on that product should bring in the UI to purchase a subscription (currently the request to report the purchase will fail)

---
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

**Comment by koke** [comment]
@toupper this isn't really working until the backend is ready, but maybe we can review, merge into trunk, and take it from there?

File: Yosemite/Yosemite/Stores/InAppPurchaseStore.swift [hunk 1]
```diff
@@ -5,8 +5,11 @@ import Networking
 
 public class InAppPurchaseStore: Store {
     private var listenTask: Task<Void, Error>?
+    private let remote: InAppPurchasesRemote
+    private var useBackend = true
```
Comment (toupper) [inline Yosemite/Yosemite/Stores/InAppPurchaseStore.swift:9]: I am wondering, what will be the usage of this property? How are we going to set it? I see that it is a private var, but I don't see any place where we change its value :)

File: Yosemite/Yosemite/Stores/InAppPurchaseStore.swift [hunk 2]
```diff
@@ -5,8 +5,11 @@ import Networking
...
 
     public override init(dispatcher: Dispatcher, storageManager: StorageManagerType, network: Network) {
+        remote = InAppPurchasesRemote(network: network)
```
Comment (toupper) [inline Yosemite/Yosemite/Stores/InAppPurchaseStore.swift:12]: Perhaps we could inject the remote so we can mock if required when testing?

**Comment by toupper** [comment]
Thanks @koke! The code looks good. Perhaps we could add some unit tests, or aren't they necessary at this point?

> @toupper this isn't really working until the backend is ready, but maybe we can review, merge into trunk, and take it from there?

Yeah, I don't see any reason why not merging that into trunk already 👍

**Review by toupper — APPROVED** [review]

File: Yosemite/Yosemite/Stores/InAppPurchaseStore.swift (hunk 1 above)
Comment (koke) [reply Yosemite/Yosemite/Stores/InAppPurchaseStore.swift:9]: Initially I wanted to make it configurable through the debug screen, but it got a bit complicated. I left that there so it was easier to change behavior during development/testing.

File: Yosemite/Yosemite/Stores/InAppPurchaseStore.swift (hunk 2 above)
Comment (koke) [reply Yosemite/Yosemite/Stores/InAppPurchaseStore.swift:12]: Looking at all the other stores in `AuthenticatedState`, I don't think we typically inject remotes, but instead allow injecting a mock `Network` to simulate network responses

**Comment by koke** [comment]
> Perhaps we could add some unit tests, or aren't they necessary at this point?

I thought about those, but I wanted to prioritize getting to a point where we could test the integration first. Also, I wasn't very confident about writing the unit tests until we could do some more manual testing. I added a note to the project to consider adding more tests later

**Pull request merged**